app/src/main/java/com/x8bit/bitwarden/data/autofill/accessibility/manager/LauncherPackageNameManagerImpl.kt

```diff
@@ -0,0 +1,43 @@
+package com.x8bit.bitwarden.data.autofill.accessibility.manager
+
+import android.content.Intent
+import android.content.pm.PackageManager
+import com.x8bit.bitwarden.data.platform.annotation.OmitFromCoverage
+import java.time.Clock
+
+/**
+ * How frequently the cached launcher list should be refreshed.
+ */
+private const val REFRESH_CACHE_MS: Long = 1L * 60L * 60L * 1000L
+
+/**
+ * The default implementation of the [LauncherPackageNameManager].
+ */
+@OmitFromCoverage
+class LauncherPackageNameManagerImpl(
+    private val clock: Clock,
+    private val packageManager: PackageManager,
+) : LauncherPackageNameManager {
+    private var lastLauncherFetchMs: Long = 0L
+    private var cachedLauncherPackages: List<String>? = null
+
+    override val launcherPackages: List<String>
+        get() {
+            if (cachedLauncherPackages == null ||
+                clock.millis() - lastLauncherFetchMs > REFRESH_CACHE_MS
+            ) {
+                updateCachedLauncherPackages()
+            }
+            return cachedLauncherPackages.orEmpty()
+        }
```

Comment:
So is this information going to be used in some way in a later PR then?

Reply:
Yep, we use this to make sure we do not attempt accessibility autofill while on the launcher.